engine/src/main/java/org/terasology/logic/players/CameraClientSystem.java

```diff
@@ -29,12 +29,30 @@
 import org.terasology.logic.location.Location;
 import org.terasology.logic.location.LocationComponent;
 import org.terasology.logic.permission.PermissionManager;
+import org.terasology.logic.players.event.OnPlayerSpawnedEvent;
 import org.terasology.logic.players.event.ResetCameraEvent;
 import org.terasology.math.geom.Quat4f;
 import org.terasology.math.geom.Vector3f;
 import org.terasology.network.ClientComponent;
 import org.terasology.registry.In;
 
+/**
+ * This is a system that creates and maintains a client side entity for the camera.
+ * <p>
+ * Entity Creation Flow:
+ * - ClientComponent for the local player added or changed
+ * - If Camera does not exist, create "engine:camera" entity
+ * - Save the ClientComponent with the new camera attached
+ * <p>
+ * Reset Camera Event Flow:
+ * - ResetCameraEvent
+ * - Camera entity on the ClientComponent is destroyed and set to EntityRef.NULL
+ * - ClientComponent is saved
+ * <p>
+ * Auto Mount Camera Flow:
+ * - ClientComponent or AutoMountCameraComponent for the local player added or changed
+ * - The local player's camera entity (created from above steps) is location linked to the Gaze Entity
+ */
 @RegisterSystem(RegisterMode.CLIENT)
```

Comment:
I feel this class-level javadoc is like the New Horizon probe swinging by Pluto: it created more questions than it answered! =D

I guess one question mark buzzing over my head as I read this javadoc is: what is actually happening in this class/system and what is happening elsewhere? The idea of describing flows is good but the actions of this system and those of collaborating code are blurred together. This could be solved by qualifying some of the statements with a subject, i.e.:

- System receives an added/changed ClientComponent event
- If the ClientComponent has no camera this system creates and attaches an "engine:camera" entity
- Finally, the system triggers the saving of the ClientComponent.

Notice how in this example I try to clarify who is doing what. I.e. in the first line it's clear it is not the system that added or changed the ClientComponent - the system merely listens for such change. Then, in the third line by using the word "triggers" I imply that this system doesn't actually do the saving, it is collaborating code that does.

The other thing I'd like to mention is that by reading this javadoc I wonder: in what cases these flows arise? Perhaps this is detailed elsewhere in the codebase and nobody would ever see this class out of context like I am doing now. But as I am reading this out of context, I do wonder: in what circumstance the camera is created? And in what circumstances does it get reset?